File: HanoiTower/hanoi.c
#include <stdio.h>

void hanoi(int n, char from, char to, char aux) {
    if (n == 1) {
        printf("Move disk 1 from %c to %c\n", from, to);
        return;
    }
    hanoi(n - 1, from, aux, to);
    printf("Move disk %d from %c to %c\n", n, from, to);
    hanoi(n - 1, aux, to, from);
}

int main() {
    int n;
    // 사용자 입력 및 유효성 검사
    do {
        printf("Enter the number of disks (positive integer): ");
        scanf("%d", &n);
        if (n <= 0) {
            printf("Please enter a positive integer.\n");
        }
    } while (n <= 0); // 유효하지 않은 경우 반복

    printf("Moves to solve the Towers of Hanoi with %d disks:\n", n);
    hanoi(n, 'A', 'C', 'B'); // A, B, C는 기둥 이름
    return 0;
}

File: Hanoi/Hanoi.py

def hanoi(n, from_rod, to_rod, aux_rod):
    # 하노이탑 문제를 해결하는 재귀 함수
    if n == 1:
        print(f"Move disk 1 from {from_rod} to {to_rod}")
        return
    hanoi(n - 1, from_rod, aux_rod, to_rod)
    print(f"Move disk {n} from {from_rod} to {to_rod}")
    hanoi(n - 1, aux_rod, to_rod, from_rod)

if __name__ == "__main__":
    while True:
        n = int(input("Enter the number of disks (positive integer): "))
        if n > 0:
            break
        print("Please enter a positive integer.")

    print(f"Moves to solve the Towers of Hanoi with {n} disks:")
    hanoi(n, 'A', 'C', 'B')  # A, B, C는 기둥 이름
